lib/Driver/ToolChain.cpp

```diff
@@ -329,5 +330,5 @@ ToolChain::constructBatchJob(ArrayRef<const Job *> unsortedJobs,
       *batchCJA, inputJobs.takeVector(), std::move(output), executablePath,
       std::move(invocationInfo.Arguments),
       std::move(invocationInfo.ExtraEnvironment),
-      std::move(invocationInfo.FilelistInfos), sortedJobs);
+      std::move(invocationInfo.FilelistInfos), sortedJobs, quasiPIDBase);
```

Comment:
>  it is now critical that this is the only place that constructs a BatchJob

No, theoretically other locations could construct BatchJobs: quasiPIDBase is just passed through to the BatchJob constructor here. This is just a wrapper for that constructor and the setup for calling into other ToolChain methods.

Reply:
Yes, but since QPids have to be unique, I would want the nextQPidBase to be maintained in this function, and for this function to be the only one constructing batch jobs. That way, the invariant would be centralized.

Reply:
Moved this logic to BatchJob.